activemodel/lib/active_model/attribute_methods.rb

```diff
@@ -489,6 +489,13 @@ def respond_to?(method, include_private_methods = false)
       end
     end
 
+    # Provide the pattern matching interface for matching against hash patterns.
+    def deconstruct_keys(keys)
+      keys.each_with_object({}) do |key, deconstructed|
+        deconstructed[key] = _read_attribute(key) if attribute_method?(key.to_s)
+      end
```

Comment:
```suggestion
      deconstructed = {}
      
      keys.each do |key|
        string_key = key.to_s
        deconstructed[key] = _read_attribute(string_key) if attribute_method?(string_key)
      end
      
      deconstructed
```

Hey @kddnewton, this is really cool! I have two small suggestions:
- use `_read_attribute(key.to_s)` instead of `_read_attribute(key)`
- use `each` instead of `each_with_object`

Active Record overrides `_read_attribute` here: https://github.com/rails/rails/blob/377ae3cee66f94d51cc7478a89eeee663173f441/activerecord/lib/active_record/attribute_methods/read.rb#L37-L39 so `key` needs to be converted to a string because `@attributes` uses string keys. (Adding a test for Active Record pattern matching is also probably a good idea)

I created a gist here to show the difference in `each` vs `each_with_object` (about an 11% speedup): https://gist.github.com/SkipKayhil/0bbee9df5e2a57bc13ece4a9db353b5f

Reply:
Nice thanks!